onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -41,6 +43,17 @@ NonCudaOps non_cuda;
 using namespace ::onnxruntime::common;
 namespace onnxruntime {
 
+// minimal KernelDef based on MetaDef instead of a Function based node
+static KernelDefBuilder& BuildFusedKernelDef(KernelDefBuilder& builder, const IndexedSubGraph::MetaDef& metadef,
+                                             const std::string& provider_type) {
+  builder.SetName(metadef.name)
+      .SetDomain(metadef.domain)
+      .SinceVersion(metadef.since_version)
+      .Provider(provider_type);
+  return builder;
+}
```

Comment:
This part is kind of confusing, why is it return the same ref back?

Reply:
The return value isn't used. Will change to return void.

Reply:
Note that we can't call Build() here to finalize as on one code path we call the kernel registry's Register method with the builder.